gio-subclass/tests/simple_application.rs

```diff
@@ -161,8 +161,6 @@ impl ops::Deref for SimpleApplication {
 fn test_arguments() {
     let app = SimpleApplication::new("org.gtk.TestApplication", gio::ApplicationFlags::empty()).unwrap();
 
-    // run is part of the trait ApplicationExtManual, so not available here!
-    // TODO: figure out how to implement that!
-    app.run(["--local"]);
+    app.run(&["--local".to_string()]);
```

Comment:
I commented on the wrong commit, sorry :) Doesn't this run forever until something quits the main loop? If not, why not here? Should the local_commandline virtual method call something to run the main loop (i.e. its parent class' implementation of it)?

Reply:
Yeah you definitely want to chain up to the parent class` implementation, it does quite some things :)

Reply:
If `local_commandline` returns 0, it will call `g_application_register()` and then `g_application_call_command_line()`. Which then calls into the `command_line` virtual method, which in your case then will give a critical warning because you don't actually implement it (you should implement it if you implement `local_commandline` from what I understand)

I believe in there you would then run the main loop, but not sure. If you don't do the custom commandline handling (i.e. don't return 0 from `local_commandline`), then `run` will run the main loop.